File: apps/admin/src/types/session.ts
import type { ChatItem } from "@/types/chat-item";
import type { NpcSummary } from "@/types/npc";

/**
 * 会话列表项，不含完整聊天记录。
 */
export interface SessionSummary {
  id: number;
  title: string;
  description: string | null;
  chat_item_count: number;
  peer_npc_ids: number[];
  self_npc_id: number | null;
  mobile_enabled: boolean;
  created_at: string;
  updated_at: string;
}

/**
 * 会话详情，包含完整聊天记录。
 */
export interface SessionDetail extends SessionSummary {
  chat_items: ChatItem[];
  peer_npcs: NpcSummary[];
  self_npc: NpcSummary | null;
}

/**
 * 创建或更新会话时的表单载荷。
 */
export interface SessionFormPayload {
  title: string;
  description: string | null;
  peer_npc_ids?: number[];
  self_npc_id?: number | null;
  chat_items?: ChatItem[];
}

File: apps/api/app/schemas/session.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.chat_item import ChatItem


class SessionSummary(BaseModel):
    """会话列表项，不含完整聊天记录。"""

    id: int = Field(description="会话 ID")
    title: str = Field(description="会话标题")
    description: str | None = Field(default=None, description="会话描述")
    chat_item_count: int = Field(description="聊天记录条数")
    created_at: datetime = Field(description="创建时间")
    updated_at: datetime = Field(description="更新时间")


class SessionDetail(SessionSummary):
    """会话详情，包含完整聊天记录 JSON。"""

    chat_items: list[ChatItem] = Field(description="聊天记录数组")


class SessionCreate(BaseModel):
    """创建会话请求体。"""

    title: str = Field(min_length=1, max_length=200, description="会话标题")
    description: str | None = Field(default=None, description="会话描述")
    chat_items: list[ChatItem] = Field(default_factory=list, description="聊天记录数组")


class SessionUpdate(BaseModel):
    """更新会话请求体，字段均为可选。"""

    title: str | None = Field(default=None, min_length=1, max_length=200, description="会话标题")
    description: str | None = Field(default=None, description="会话描述")
    chat_items: list[ChatItem] | None = Field(default=None, description="聊天记录数组")
